lib/fastlane/actions/resize_app_icon.rb

```diff
@@ -0,0 +1,120 @@
+              previous_icon = image['filename']
+              if previous_icon
+                previous_icon_file = File.join(@icon_set, previous_icon)
+                if File.exist?(previous_icon_file) and !scaled_image_name.eql?(previous_icon)
+                  FileUtils.rm(previous_icon_file)
+                end
+              end
+
+              image['filename'] = scaled_image_name
+
+            else
+              Helper.log.info 'Failed during the conversion process for some reason... 😱' and abort
+            end
+          end
+
+          # Output the new JSON contents
+          File.open(@contents_file, "w") do |f|
+            f.write(JSON.pretty_generate(contents))
+          end
+
+          Helper.log.info 'All icons built and installed 👍'
+        end
+      end
+
+      def self.validate_image_magick!
+        abort('You need to install Image Magick! Check http://www.imagemagick.org for instructions.') unless system("which convert > /dev/null 2>&1")
+      end
+
+      def self.validate_icon!
+        @icon = File.expand_path(@icon)
+        abort("Oops! Can't find the source icon you specified. #{@icon}") unless File.exist?(@icon)
+      end
+
+      def self.validate_asset_catalog!
+        @asset_catalog = File.expand_path(@asset_catalog)
+        abort("Oops! Can't find the asset catalog you specified. #{@asset_catalog}") unless File.exist?(@asset_catalog)
+        abort("Oops! It does not seem you specified a valid asset catalog.  It needs to be the .xcassets directory.") \
+        unless File.directory?(@asset_catalog) and File.extname(@asset_catalog).eql?('.xcassets')
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Resize a source 1024x1024 app icon image to all sizes in an .xcassets collection."
+      end
+
+      def self.details
+        "Relies on ImageMagick to resize a source icon image into all required image sizes as defined in the specified .xcassets collection."
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :iconPath,
+                                       env_name: "FL_RESIZE_SOURCE_ICON_PATH",
+                                       description: "The path to the original 1024x1024 image",
+                                       is_string: true, # true: verifies the input is a string, false: every kind of value
+                                       optional: false),
+          FastlaneCore::ConfigItem.new(key: :assetsPath,
+                                       env_name: "FL_RESIZE_ASSETS_PATH",
+                                       description: "The path to the xcassets folder for the app icons",
+                                       is_string: true, # true: verifies the input is a string, false: every kind of value
+                                       optional: false)
```

Comment:
How about adding a `default_value: ` here which is
```ruby
default_value: Dir["./**/AppIcon.appiconset"].last
```
which will be `nil` if it can't be found